{
  int attemptCnt=0;
  while (true) {
    try {
      attemptCnt++;
      clo.call();
      return attemptCnt;
    }
 catch (    Exception e) {
      if (attemptCnt >= attempts)       throw e;
 else       X.println("Failed to execute closure in " + attempts + " attempts.");
    }
  }
}
